apollo-federation/src/operation/rebase.rs

```diff
@@ -667,12 +672,13 @@ impl NamedFragments {
                 .get_type(fragment.type_condition_position.type_name().clone())
                 .and_then(CompositeTypeDefinitionPosition::try_from)
             {
-                if let Ok(mut rebased_selection) = fragment.selection_set.rebase_on(
+                let mut rebased_selection = fragment.selection_set.rebase_on(
                     &rebased_type,
                     &rebased_fragments,
                     schema,
                     RebaseErrorHandlingOption::IgnoreError,
-                ) {
+                )?;
+                if !rebased_selection.is_empty() {
```

Comment:
this is already handled by `is_selection_set_worth_using` below

Reply:
I reverted this change for now, since it's distracting.